splash/html_element.py

```diff
@@ -0,0 +1,317 @@
+from splash.exceptions import DOMError, JsError
+from splash.jsutils import escape_js
+from splash.casperjs_utils import (
+    VISIBLE_JS_FUNC,
+    ELEMENT_INFO_JS,
+    FIELD_VALUE_JS,
+    FORM_VALUES_JS
+)
+from functools import wraps
+
+DIMENSIONS_JS_FUNC = """
+(function(elem) {
+    var rect = elem.getClientRects()[0];
+    return {"x":rect.left, "y": rect.top, "width": rect.width, "height": rect.height}
+})(%s)
+"""
+
+FETCH_TEXT_JS_FUNC = """
+(function(elem) {
+    return (elem.textContent || elem.innerText || elem.value || '').trim();
+})(%s)
+"""
+
+FILL_FORM_VALUES_JS = """
+function fill(form, values) {
+    Object.keys(values).forEach(function(name) {
+        var input = form.querySelector("input[name='" + name + "']")
+            || form.querySelector("select[name='" + name + "']")
+            || form.querySelector("textarea[name='" + name + "']");
```

Comment:
ideally this should use CSS.escape for field names; it is a big [copy-paste](https://github.com/mathiasbynens/CSS.escape/blob/ff41bbb0702cdb4ca3e7202cc123b24bd472e1c0/css.escape.js#L21) though :(